src/realm/query_engine.cpp

```diff
@@ -51,6 +51,9 @@ size_t ParentNode::find_first(size_t start, size_t end)
 
 void ParentNode::aggregate_local_prepare(Action TAction, DataType col_id, bool nullable)
 {
+    // Verify that the cached column accessor is still valid
+    verify_column(); // throws
```

Comment:
All the `verify_column()` functions are now exercised during the "delete column" test cases.